README.md

```diff
@@ -1,67 +1,77 @@
-expected for any language supported in Bazel. The packaging rules provide
-support for integration with dependencies that, in a non-Bazel environment,
-would typically be managed by `pip`.
-
-Historically, the core rules have been bundled with Bazel itself. The Bazel
-team is in the process of transitioning these rules to live in
-bazelbuild/rules_python instead. In the meantime, all users of Python rules in
-Bazel should migrate their builds to load these rules and their related symbols
-(`PyInfo`, etc.) from `@rules_python` instead of using built-ins or
-`@bazel_tools//tools/python`.
+This repository is the home of the core Python rules -- `py_library`,
+`py_binary`, `py_test`, and related symbols that provide the basis for Python
+support in Bazel. It also contains packaging rules for integrating with PyPI
+(`pip`). Documentation lives in the
+[`docs/`](https://github.com/bazelbuild/rules_python/tree/master/docs)
+directory and in the
+[Bazel Build Encyclopedia](https://docs.bazel.build/versions/master/be/python.html).
+
+Currently the core rules are bundled with Bazel itself, and the symbols in this
+repository are simple aliases. However, in the future the rules will be
+migrated to Starlark and debundled from Bazel. Therefore, the future-proof way
+to depend on Python rules is via this repository. See[`Migrating from the Bundled Rules`](#Migrating-from-the-bundled-rules) below.
+
+The core rules are stable. Their implementation in Bazel is subject to Bazel's
+[backward compatibility policy](https://docs.bazel.build/versions/master/backward-compatibility.html).
+Once they are fully migrated to rules_python, they may evolve at a different
+rate, but this repository will still follow
+[semantic versioning](https://semver.org).
+
+The packaging rules (`pip_import`, etc.) are less stable. We may make breaking
+changes as they evolve. There are no guarantees for rules underneath the
+`experimental/` directory.
+
+## Getting started
+
+To import rules_python in your project, you first need to add it to your
+`WORKSPACE` file. If you are using the [Bazel
+Federation](https://github.com/bazelbuild/bazel-federation), you will want to
+copy the boilerplate in the rules_python release's notes, under the "WORKSPACE
+setup" heading. This will look something like the following:
 
-## Setup
+```python
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+http_archive(
+    name = "rules_python",
+    # NOT VALID: Replace with actual version and SHA.
+    url = "https://github.com/bazelbuild/rules_python/releases/download/<RELEASE>/rules_python-<RELEASE>.tar.gz",
+    sha256 = "<SHA>",
+)
+load("@rules_python//python:repositories.bzl", "py_repositories", "rules_python_toolchains")
+py_repositories()
+rules_python_toolchains()
+```
 
-To use this repository, first modify your `WORKSPACE` file to load it and call
-the initialization functions as needed:
+Otherwise, you may import rules_python in a standalone way by copying the
+following:
```

Comment:
I don't see the difference between the two blocks of code. You just replaced `http_archive` with `git_repository`? (or did I miss something?)

Reply:
It started out more different, but then converged when I fixed errors in the federation boilerplate. Now I'm at a loss to understand how exactly the federation comes into play here. @aiuto, can you comment on what in particular a user does to use the federation? It's possible I conflated using the federation with using the distributable tar.gz.

Reply:
Discussed offline with Tony. What I had is a comparison of using the distributable vs source checkout, but using the federation is different. Adding a third way to write WORKSPACE files, arranged in order of preference.